Added G-API implementation of OMZ IFD demo

aDanPin opened the pull request.

aDanPin committed:
Added G-API implementation of OMZ IFD demo

aDanPin committed:
Added G-API implementation of OMZ IFD demo

aDanPin commented:
@dmatveev 
Please review
This pull-request is a copy of the [previous](https://github.com/aDanPin/open_model_zoo/pull/5)

dmatveev commented on demos/gapi_interactive_face_detection_demo/README.md:
```diff
@@ -0,0 +1,116 @@
+# Interactive Face Detection C++ Demo
+
```
This readme is not accurate anymore

dmatveev commented on demos/gapi_interactive_face_detection_demo/README.md:
```diff
@@ -0,0 +1,116 @@
+# Interactive Face Detection C++ Demo
```
also, I'd rename directory to `interactive_face_detection_demo` to make the directory listing order correct

dmatveev commented on demos/gapi_interactive_face_detection_demo/main.cpp:
```diff
@@ -0,0 +1,567 @@
+};
+
+void rawOutputDetections(const cv::Mat &ssd_result,
+                         const cv::Size upscale,
```
`const cv::Size &`

dmatveev commented on demos/gapi_interactive_face_detection_demo/main.cpp:
```diff
@@ -0,0 +1,567 @@
+    }
+}
+
+void faceDataUpdate(cv::Mat frame,
```
`const cv::Mat &`?

dmatveev commented on demos/gapi_interactive_face_detection_demo/main.cpp:
```diff
@@ -0,0 +1,567 @@
+
+void faceDataUpdate(cv::Mat frame,
+                    Face::Ptr &face,
+                    cv::Rect face_rect,
```
`const cv::Rect &`

dmatveev commented on demos/gapi_interactive_face_detection_demo/main.cpp:
```diff
@@ -0,0 +1,567 @@
+                    std::vector<cv::Rect> &face_hub,
+                    size_t &id,
+                    bool no_smooth) {
+    // Face apdate
```
"apdate"?

dmatveev commented on demos/gapi_interactive_face_detection_demo/main.cpp:
```diff
@@ -0,0 +1,567 @@
+                cv::GMat in;
+
+                cv::GMat frame = cv::gapi::copy(in);
+                cv::GProtoOutputArgs outs = GOut(frame);
```
maybe `auto outs = ...` is better here. We don't really expose this type so much in our API.

dmatveev commented on demos/gapi_interactive_face_detection_demo/main.cpp:
```diff
@@ -0,0 +1,567 @@
+                                                          FLAGS_bb_enlarge_coef,
+                                                          FLAGS_dx_coef,
+                                                          FLAGS_dy_coef);
+                outs += GOut(faces);
```
The above three operations are unconditional, so you could do it this way instead of multiple `+=`s:

```c++
auto outs = cv::GOut(frame, detections, faces);
```

And then do your `+=` where the outputs are conditional.

dmatveev commented on demos/gapi_interactive_face_detection_demo/main.cpp:
```diff
@@ -0,0 +1,567 @@
+        out_vector += cv::gout(ssd_res);
+
+        std::vector<cv::Rect> face_hub;
+        out_vector += cv::gout(face_hub);
```
same thing here - your three outputs are unconditional so you could do

```c++
auto out_vector = cv::gout(frame, ssd_res, face_hub);
```

BTW what is `hub`?

dmatveev commented on demos/gapi_interactive_face_detection_demo/main.cpp:
```diff
@@ -0,0 +1,567 @@
+                                                                "happy",
+                                                                "sad",
+                                                                "surprise",
+                                                                "anger"});
```
please move the vector of emotions to some global static constant variable.

dmatveev commented on demos/gapi_interactive_face_detection_demo/main.cpp:
```diff
@@ -0,0 +1,567 @@
+                framesCounter++;
+                timer.finish("total");
+            } else { // End of streaming
+                std::cout<<"End of streaming" << std::endl;
```
Fix alignment in `cout<<"`

dmatveev commented on demos/gapi_interactive_face_detection_demo/utils.hpp:
```diff
@@ -0,0 +1,97 @@
+    float angle_y;
+    float angle_p;
+    float angle_r;
+};
```
Do you need these three structures in your code?

dmatveev reviewed (commented):
Few minor comments.

Looks got to me, looks worth proposing for OMZ now. Well done!

aDanPin committed:
Small refactoring

aDanPin committed:
Small refactoring

aDanPin committed:
Fix readme

aDanPin committed:
Fix readme

aDanPin committed:
Fix readme

aDanPin committed:
Fix readme

aDanPin committed:
Update demos/interactive_face_detection_demo_gapi/CMakeLists.txt

Co-authored-by: Zlobin Vladimir <vladimir.zlobin@intel.com>

aDanPin committed:
Update demos/interactive_face_detection_demo_gapi/CMakeLists.txt

Co-authored-by: Zlobin Vladimir <vladimir.zlobin@intel.com>

aDanPin committed:
Update demos/interactive_face_detection_demo_gapi/CMakeLists.txt

Co-authored-by: Zlobin Vladimir <vladimir.zlobin@intel.com>

aDanPin committed:
Update demos/interactive_face_detection_demo_gapi/CMakeLists.txt

Co-authored-by: Zlobin Vladimir <vladimir.zlobin@intel.com>

aDanPin committed:
Update demos/interactive_face_detection_demo_gapi/README.md

Co-authored-by: Zlobin Vladimir <vladimir.zlobin@intel.com>

aDanPin committed:
Update demos/interactive_face_detection_demo_gapi/README.md

Co-authored-by: Zlobin Vladimir <vladimir.zlobin@intel.com>

aDanPin committed:
Update demos/interactive_face_detection_demo_gapi/main.cpp

Co-authored-by: Zlobin Vladimir <vladimir.zlobin@intel.com>

aDanPin committed:
Update demos/interactive_face_detection_demo_gapi/main.cpp

Co-authored-by: Zlobin Vladimir <vladimir.zlobin@intel.com>

aDanPin committed:
Update demos/interactive_face_detection_demo_gapi/main.cpp

Co-authored-by: Zlobin Vladimir <vladimir.zlobin@intel.com>

aDanPin committed:
Update demos/interactive_face_detection_demo_gapi/main.cpp

Co-authored-by: Zlobin Vladimir <vladimir.zlobin@intel.com>

aDanPin committed:
Update demos/interactive_face_detection_demo_gapi/main.cpp

Co-authored-by: Zlobin Vladimir <vladimir.zlobin@intel.com>

aDanPin committed:
Update demos/interactive_face_detection_demo_gapi/main.cpp

Co-authored-by: Zlobin Vladimir <vladimir.zlobin@intel.com>

aDanPin committed:
Update demos/interactive_face_detection_demo_gapi/main.cpp

Co-authored-by: Zlobin Vladimir <vladimir.zlobin@intel.com>

aDanPin committed:
Update demos/interactive_face_detection_demo_gapi/main.cpp

Co-authored-by: Zlobin Vladimir <vladimir.zlobin@intel.com>

aDanPin committed:
Update demos/interactive_face_detection_demo_gapi/main.cpp

Co-authored-by: Zlobin Vladimir <vladimir.zlobin@intel.com>

aDanPin committed:
Update demos/interactive_face_detection_demo_gapi/main.cpp

Co-authored-by: Zlobin Vladimir <vladimir.zlobin@intel.com>

aDanPin committed:
Update demos/interactive_face_detection_demo_gapi/main.cpp

Co-authored-by: Zlobin Vladimir <vladimir.zlobin@intel.com>

aDanPin committed:
Update demos/interactive_face_detection_demo_gapi/main.cpp

Co-authored-by: Zlobin Vladimir <vladimir.zlobin@intel.com>

aDanPin committed:
Update demos/interactive_face_detection_demo_gapi/main.cpp

Co-authored-by: Zlobin Vladimir <vladimir.zlobin@intel.com>

aDanPin committed:
Update demos/interactive_face_detection_demo_gapi/main.cpp

Co-authored-by: Zlobin Vladimir <vladimir.zlobin@intel.com>

aDanPin committed:
Update demos/interactive_face_detection_demo_gapi/main.cpp

Co-authored-by: Zlobin Vladimir <vladimir.zlobin@intel.com>

aDanPin committed:
Update demos/interactive_face_detection_demo_gapi/main.cpp

Co-authored-by: Zlobin Vladimir <vladimir.zlobin@intel.com>

aDanPin committed:
Update demos/interactive_face_detection_demo_gapi/main.cpp

Co-authored-by: Zlobin Vladimir <vladimir.zlobin@intel.com>

aDanPin committed:
Update demos/interactive_face_detection_demo_gapi/main.cpp

Co-authored-by: Zlobin Vladimir <vladimir.zlobin@intel.com>

aDanPin committed:
Update demos/interactive_face_detection_demo_gapi/main.cpp

Co-authored-by: Zlobin Vladimir <vladimir.zlobin@intel.com>

aDanPin committed:
Update demos/interactive_face_detection_demo_gapi/main.cpp

Co-authored-by: Zlobin Vladimir <vladimir.zlobin@intel.com>

aDanPin committed:
Review response

aDanPin committed:
Review response

aDanPin committed:
Review response

	modified:   demos/common/monitors/include/monitors/presenter.h

aDanPin committed:
Review response

	modified:   demos/common/monitors/include/monitors/presenter.h

aDanPin committed:
Add removed file

aDanPin committed:
Add removed file

aDanPin committed:
Review response

aDanPin committed:
Review response

aDanPin committed:
Refactoring

aDanPin committed:
Refactoring

Commit:
Fix video capturing

Commit:
Fix video capturing

aDanPin committed:
Update demos/interactive_face_detection_demo_gapi/interactive_face_detection_gapi.hpp

Co-authored-by: Roman Donchenko <roman.donchenko@intel.com>

aDanPin committed:
Update demos/interactive_face_detection_demo_gapi/interactive_face_detection_gapi.hpp

Co-authored-by: Roman Donchenko <roman.donchenko@intel.com>

aDanPin committed:
Update demos/interactive_face_detection_demo_gapi/README.md

Co-authored-by: Roman Donchenko <roman.donchenko@intel.com>

aDanPin committed:
Update demos/interactive_face_detection_demo_gapi/README.md

Co-authored-by: Roman Donchenko <roman.donchenko@intel.com>

aDanPin committed:
Update demos/interactive_face_detection_demo_gapi/README.md

Co-authored-by: Roman Donchenko <roman.donchenko@intel.com>

aDanPin committed:
Update demos/interactive_face_detection_demo_gapi/README.md

Co-authored-by: Roman Donchenko <roman.donchenko@intel.com>

aDanPin committed:
Review response

aDanPin committed:
Review response

dmatveev commented:
I believe this particular PR can be closed? (Given that you're moving to the main OpenVINO Github now)

dmatveev commented:
Please close this PR

vladimir-dudnik committed:
remove deprecated face-detection-adas-binary model from models.lst

vladimir-dudnik committed:
remove deprecated face-detection-adas-binary model from models.lst

vladimir-dudnik committed:
remove ref to download01.org

vladimir-dudnik committed:
remove ref to download01.org

Commit:
Review response

Commit:
Review response

Commit:
Fix unused imports in Python code

Either by removing them, or by marking them as used with __all__.

Commit:
flake8: enable unused import checks

vladimir-dudnik committed:
add tests

Commit:
Remove all star imports

Star imports prevent flake8 from finding undefined names, and can cause
situations where a module is accidentally imported indirectly through another
module (as demonstrated by demos/tests/cases.py and collections).

Commit:
flake8: enable all remaining pyflakes checks

Commit:
Merge pull request #1878 from IRDonch/flake8-deprecations

flake8: enable the pycodestyle deprecation warnings

Wovchena committed:
demos/classification: move PerformanceMetrics out of GridMat, remove frame arg from textUpdate()

Commit:
Review response

Commit:
Review response

eaidova committed:
AC: offline mode upgrade (#1879)

* AC: offline mode upgrade

* async mode

* progress tracking

* autonom adapter output selection

* iterative loading and dumping

* errors and warnings

* add tests

vladimir-dudnik committed:
add classes description

Commit:
Merge pull request #1873 from IRDonch/od-pathlib

Switch object_detection_demo and common/models to pathlib

Commit:
Merge pull request #1876 from Wovchena/use-unified-perf-metrics-in-demos

Use perf metrics in classification_demo

anzhella-pankratova committed:
Separe VideoCapWrapper into classes VideoCapWrapper and CameraCapWrapper

anzhella-pankratova committed:
Merge branch 'master' of https://github.com/anzhella-pankratova/open_model_zoo

vladimir-dudnik committed:
Merge pull request #1882 from vladimir-dudnik/vd/fix-readme-person-vehicle-bike-detection-crossroad-0078

add classes description

Commit:
demos: clean up the compiler options

* Use the modern CMake commands for adding compiler options.
* Remove redundant options:
  * /EHsc is supplied by default;
  * the various -W options are included in -Wall;
  * -std=c++11 isn't needed, because we now require a version of CMake that
    supports CMAKE_CXX_STANDARD.
  * /LARGEADDRESSAWARE only makes sense on 32-bit Windows, which is not
    supported.
* Distinguish between Windows-specific and MSVC-specific options.
* Enable -Wall for Clang and not just for GCC.

eaidova committed:
AC: decrease numpy version requirement for core installation (#1883)

Commit:
demos: fix warnings found with Clang

* inconsistent override qualifiers;
* unnecessary std::move;
* unused class member.

Commit:
Merge pull request #1884 from IRDonch/clean-compiler-options

demos: clean up the compiler options

Commit:
Update gflags to version 2.2.2

Commit:
demos: group compiler flags together

This is now possible, because gflags 2.2.2 fixed enough warnings that
it's now buildable with -Wall.

Commit:
flake8: enable some pycodestyle warnings

Commit:
Merge pull request #1885 from IRDonch/clang-warnings

demos: fix warnings found with Clang

anzhella-pankratova committed:
Delete loop flag from Base class and CameraCapWrapper

Commit:
AC: Update accuracy evaluation method for facenet model (#1887)

* Update accuracy evaluation method for facenet model

* Fix comments

Commit:
Merge pull request #1886 from IRDonch/gflags-update

Update gflags to version 2.2.2

anzhella-pankratova committed:
Delete __init__() in class ImagesCapture

Commit:
Merge pull request #1815 from anzhella-pankratova/master

Add generalized reader of images and video for python demos

anzhella-pankratova committed:
Add path to import module in video.py

eaidova committed:
AC: fix text spotting quantization (#1892)

Commit:
Switch a few Python demos from os.path to pathlib

Commit:
Rename variables named "l" to less ambiguous names

Wovchena committed:
demos/classification, segmentation: guess nireq, minor

Commit:
AC: Simplify some code

Commit:
AC: Added support for CRAFT model (#1725)

* Added support for CRAFT model

Commit:
Merge pull request #1895 from IRDonch/ac-simplify

AC: Simplify some code

Commit:
Model renamed

anzhella-pankratova committed:
Check if tests for image_retrieval_demo catch an error or not

Commit:
machine_translation_demo: use string formatting instead of concatenation

Commit:
Merge pull request #1896 from IRDonch/more-pathlib

Switch a few Python demos from os.path to pathlib

anzhella-pankratova committed:
Create SegmentationModel class

anzhella-pankratova committed:
Delete print

anzhella-pankratova committed:
Change main.py in segmentation demo

Wovchena committed:
Update demos/common/models/src/detection_model.cpp

Co-authored-by: Fedor Zharinov <fedor.zharinov@intel.com>

Wovchena committed:
demos: Default value is -> If this option is omitted, number of infer requests is

Wovchena committed:
demos: remove trailing whitespace

anzhella-pankratova committed:
Final version of image_retrieval_demo

Wovchena committed:
demos: replace std::cerr

Wovchena committed:
demos/async_pipeline: shift indentation one level to the left

Commit:
Merge pull request #1894 from Wovchena/demos-guess-nireq-minor

demos/classification, segmentation: guess nireq, minor

eaidova committed:
AC: align hour glass support with original implementation (#1899)

anzhella-pankratova committed:
Change __init__ and postprocess in SegmentationModel, modify main.py

anzhella-pankratova committed:
Remove -loop options

anzhella-pankratova committed:
Rewrite message showed by RuntimeError

Commit:
Merge pull request #1900 from openvinotoolkit/release

Merge release to develop

anzhella-pankratova committed:
Change input in segmentation_py tests

anzhella-pankratova committed:
Delete _vector in variables names

Commit:
Specify the exception type in every except block

Commit:
flake8: enable most of the statement style checks

I only ignored the ones about multiple statements on the same line, because
I think in some cases that looks fine and it saves vertical space.

Commit:
AC: remove an unnecessary repeated import

Commit:
Fix comments

Wovchena committed:
demos/classification: common pipeline

Commit:
Merge pull request #1857 from Anna-Mironova/am/support-ssh

Added SSH model

Commit:
Merge pull request #1874 from IRDonch/unused-imports

flake8: add a check for unused imports

Commit:
Merge pull request #1901 from IRDonch/flake8-style

flake8: enable most statement style checks

anzhella-pankratova committed:
Change default colormap, add capability of applying custom colormap

Commit:
Review response

Commit:
Review response

anzhella-pankratova committed:
Merge branch 'develop' into segmentation_py

anzhella-pankratova committed:
Delete old segmentation_demo

anzhella-pankratova committed:
Rename: segmentation_demo_async -> segmentation_demo

anzhella-pankratova committed:
Update CMakeLists.txt, README.md, models.lst

anzhella-pankratova committed:
Add auto_resize argument

Commit:
Insert missing spaces

Commit:
Add blank lines where needed

Commit:
flake8: enable select whitespace checks from pycodestyle

pycodestyle has a plethora of these, however, most of them fall into one
or more of the following categories:

* too situational (e.g. extra spaces are usually wrong, but sometimes they
  are used to format code as a table);
* not important enough (e.g. missing spaces around operators);
* producing enough warnings on our code that I can't be bothered to fix them.

Still, several of the checks are usable, so enable those.

anzhella-pankratova committed:
Resolve conflict

anzhella-pankratova committed:
Delete frame_size variable from whiteboard_inpainting_demo

anzhella-pankratova committed:
Remove unused imports

anzhella-pankratova committed:
Rewrite README.md

anzhella-pankratova committed:
Update tests for segmentation_demo.py, add small fixes from code review

anzhella-pankratova committed:
Delete extra brackets

eaidova committed:
remove old parameters serialization API (#1905)

* remove parameters serialization API

* remove topology type

anzhella-pankratova committed:
Resolve conflicts

anzhella-pankratova committed:
Undo changes in image_retrieval_demo

anzhella-pankratova committed:
Set default colormap as pascal voc colormap

Commit:
Merge pull request #1848 from anzhella-pankratova/demos_omz

Update reading of input data in python demos

Commit:
Merge pull request #1890 from anzhella-pankratova/segmentation_cpp

Replace segmentation_demo by segmentation_demo_async

Commit:
Remove unneeded spaces

Commit:
Collapse spaces

anzhella-pankratova committed:
Modify tests for demo

Commit:
Indent closing brackets same as the line with the opening bracket

Commit:
AC: replace a cumbersome conditional expression with an if

Commit:
Merge pull request #1904 from IRDonch/flake8-whitespace

Fix spacing errors and enable spacing-related pycodestyle checks

Commit:
flake8: enable check for closing bracket indentation

Commit:
Merge pull request #1903 from IRDonch/flake8-star-imports

flake8: enable all remaining pyflakes checks

anzhella-pankratova committed:
Expand available models list

Commit:
run_tests.py: explicitly pass the modified environment to demos

This makes it more clear that we're running the demos with a modified
environment, and ensures we're not modifying the environment of any
non-demo child process.

Commit:
Merge pull request #1909 from IRDonch/demo-env

run_tests.py: explicitly pass the modified environment to demos

Wovchena committed:
demos/classification: gflags::uint32 -> unsigned int, update error messages

Commit:
Merge pull request #1910 from IRDonch/flake8-brackets

flake8: enable check for closing bracket indentation

anzhella-pankratova committed:
Create Visualizer class, modify models.lst

anzhella-pankratova committed:
Change default color_palette in segmentation demo

anzhella-pankratova committed:
Small fixes

anzhella-pankratova committed:
Delete unused import

anzhella-pankratova committed:
Changes from code review

anzhella-pankratova committed:
Add extra indentation

Wovchena committed:
demos: fix docstrings, add override, remove virtual keywords

anzhella-pankratova committed:
Small fix

Commit:
Merge pull request #1913 from anzhella-pankratova/segmentation_cpp

Change default colormap in C++ segmentation demo

Commit:
Merge pull request #1902 from Wovchena/classification_demo_common_pipeline

demos/classification: common pipeline

Commit:
tools/downloader/requirements-pytorch.in: restrict the torchvision version

torchvision releases are bound to the corresponding releases, so if we need
PyTorch<1.7, we also need torchvision<0.8. This change isn't actually needed
for pip, because the new resolver introduced in pip 20.3 is capable of picking
the correct torchvision version, however:

* pip-compile can't use the new resolver yet, so it fails to compile the
  requirements file if torchvision isn't restricted; and

* pip works faster if you tell it from the start which version of torchvision
  to use.

Commit:
Bump all dependency versions

Wovchena committed:
demos: fix/disable MSVC warnings

The compiler asks to have `/EHsc` if cmake is run with
`-DCMAKE_CXX_FLAGS=/WX`.
`WIN32` is not defined.
`fileNameNoExt()` is unused from anywhere.
Disable warnings for `ctcdecode_numpy_impl` for MSVC as well as for GCC
like compilers.

Commit:
Merge pull request #1915 from IRDonch/bump-requirements

Bump all dependency versions

Commit:
cocosnet: fix conversion when a separate conversion directory is used

Commit:
cocosnet: fix a deprecation warning produced during conversion

Commit:
cocosnet: patch the source files to make them convertable with PyTorch 1.8+

Previously, PyTorch used to always emit the ONNX Squeeze op with fixed
dimensions. Now, however, it will emit a dimensionless Squeeze when the
squeeze function is called without the dim argument.

MO can't handle dimensionless Squeeze operations (yet, anyway). Work around
it by patching the model source to add a dim argument.

anzhella-pankratova committed:
Fixes from code review

anzhella-pankratova committed:
Merge branch 'develop' into segmentation_py

Wovchena committed:
demos: don't rely on /W3 being set by default, be specific in warning supression for ctcdecode_numpy_impl

Commit:
Merge pull request #1897 from anzhella-pankratova/segmentation_py

Unify Python segmentation demo

Commit:
Model densenet-121-tf сhanged to tf2

Commit:
Model densenet-169-tf сhanged to tf2

Commit:
Model densenet-201-tf from tf2 added

Commit:
Fix misprints

Commit:
Merge pull request #1919 from IRDonch/cocosnet-fixes

Several fixes for cocosnet

Commit:
Enable a few more yamllint checks

It can be useful to have more than one space after a colon in order to align
map values (and this is heavily used in existing configs), so set
max-spaces-after to -1 to disable that check.

Wovchena committed:
demos: fix C4834 warning

Commit:
Merge pull request #1922 from IRDonch/more-yamllint

Enable a few more yamllint checks

Wovchena committed:
demos: manage word_ptrs's size

Commit:
ci/requirements-demos.txt: downgrade NumPy to work around a Windows bug

See https://github.com/numpy/numpy/issues/16744. Hopefully, this bug will
be fixed in Windows by the time we bump the requirements next time, so we
won't need to permanently pin the NumPy version.

Commit:
Merge pull request #1923 from IRDonch/fix-requirements

ci/requirements-demos.txt: downgrade NumPy to work around a Windows bug

Commit:
Merge pull request #1918 from Wovchena/demos-fix-disable-MSVC-warnings

demos: fix/disable MSVC warnings

Commit:
Added configs for colorization models without pipeline (#1914)

* Added configs for colorization models without pipeline

* Added subset of images, updated docs

Commit:
Input parameter message fix

Commit:
Bump the version for the next release

vladimir-dudnik committed:
object_detection_demo models.lst (#1916)

* extend object_detection_demo models.lst with ssd models

* put items in alphabet order

Commit:
Rename APACHE-2.0-TensorFlow.txt to APACHE-2.0-TF-Models.txt, update references

Commit:
Merge pull request #1926 from IRDonch/bump-version

Bump the version for the next release

Commit:
Merge pull request #1931 from Anna-Mironova/am/update-license

APACHE-2.0-TensorFlow.txt renamed to APACHE-2.0-TF-Models.txt, updated references

Commit:
Merge pull request #1928 from fzhar/develop

Input parameter message fix

eaidova committed:
AC: fix input feeding for caffe launcher (#1932)

eaidova committed:
AC: removing numpy 1.18 restriction

Commit:
Fix comments

Commit:
Added APACHE-2.0-TensorFlow.txt to match the license from tf repo

eaidova committed:
Update requirements-ac-test.txt

eaidova committed:
Update requirements-ac.txt

Commit:
Merge pull request #1920 from Anna-Mironova/am/update-densenet

Changed densenet-tf models to tf2

Commit:
Merge pull request #1934 from eaidova/ea/numpy_restriction_gone

AC: removing numpy 1.18 restriction

Commit:
Updated description for SSD models (#1935)

eaidova committed:
AC: fix relative path specification for saved model directory (#1933)

Commit:
Added ReXNet v1 1.0x model (#1906)

* Added ReXNet v1 1.0x model

* Added GFlops and mParams

* Fix comments

eaidova committed:
demos: explicitly specify supported model for human-pose-estimation demo (#1936)

anzhella-pankratova committed:
Python demos: add .gif files to README (#1925)

* Add human_pose_estimation_3d.gif

* Delete .jpg file (replaced by gif)

* Add human_pose_estimation.gif

* Add instance_segmentation.gif

* Add object_detection.gif

* Add segmentation.gif

* Add single_human_pose_estimation.gif

* Add image_retrieval.gif

* Increase image_retrieval.gif size

* Label detections on object_detection.gif

anzhella-pankratova committed:
Add models to object_detection_demo models.lst (#1927)

* Extend models.lst for Python demo

* Extend models.lst for C++ demo

* add product-detection-0001 to object_detection_demo

* remove ssd-resnet34-1200

Co-authored-by: Vladimir Dudnik <vladimir.dudnik@intel.com>

Commit:
Review response

Commit:
Review response

Commit:
Review response

Commit:
Review response

Commit:
Review response

Commit:
Review response

vladimir-dudnik committed:
update ssd models description (#1938)

* add details on dataset

* fix voc2012 link

* add output values range

* fix backround typo

* Update models/public/ssd300/ssd300.md

ok, thanks

Co-authored-by: Ekaterina Aidova <ekaterina.aidova@intel.com>

* remove background, which is suppressed by ssd DetectionOutput

* polishing phrazes and add ssd512

Co-authored-by: Ekaterina Aidova <ekaterina.aidova@intel.com>

eaidova committed:
AC: use cached anchor boxes to speed up evaluation (#1943)

* AC: use cached anchor boxes to speed up evaluation

* fix top k parameter in retinaface adapter

ivanvikhrev committed:
cpp object detection demo: add faceboxes wrapper (#1940)

* add faceboxes wrapper class draft

* add featuremaps

* add featuremaps2

* add anchors calculation

* star impl of postprocessing

* add anchors class

* complete postrocessing

* minor fixes, cleaning code

* upd models.lst

* delete trailing spaces, change models order in -at

* fix warnings

* switch to int and float instead double

* convert rest of doubles to float

Co-authored-by: Vikhrev <ivan.vikhrev@intel.com>
Co-authored-by: Vladimir Dudnik <vladimir.dudnik@intel.com>

Commit:
Remove labels list adjustment in object detection models (SSD and Yolo). (#1941)

* Remove labels list adjustment in object detection models (SSD and Yolo)

* Update demos/object_detection_demo/README.md

Co-authored-by: Vladimir Dudnik <vladimir.dudnik@intel.com>

Co-authored-by: Vladimir Dudnik <vladimir.dudnik@intel.com>

vladimir-dudnik committed:
main.cpp: remove trailing spaces

vladimir-dudnik committed:
suppress warnings, get back removed fileNameNoExt func

anzhella-pankratova committed:
C++ demos: add .gif files to README (#1939)

* Add Python action_recognition.gif

* Add .gif files for C++ demos

* Rename security_barrier_camera.gif

* Add smart_classroom.gif

* Show more capabilities of interactive_face_detection demo by .gif

* Add Python whiteboard_inpainting.gif

* Change whiteboard_inpainting.gif

eaidova committed:
AC: sipport camvid with 32 classes and segmentation model with flatten out (#1946)

vladimir-dudnik committed:
rfcn-resnet101-coco-tf in object_detection_demo (#1947)

* add rfcn to object_detection_demo models.lst

* add rfcn-resnet101-coco-tf test cases

eaidova committed:
AC: added info about dumped annotation (#1948)

* AC: added info about dumped annotation

* update tests

* bump version

Commit:
Review response

Commit:
Review response

Commit:
Review response

vladimir-dudnik committed:
merge aDanPin changes

vladimir-dudnik committed:
fix trailing spaces

vladimir-dudnik committed:
fix unreferenced variable

vladimir-dudnik committed:
fix missed required param message

Commit:
Review response

Commit:
Merge branch 'dp/gapi_interactive_face_detection_demo' of https://github.com/aDanPin/open_model_zoo into dp/gapi_interactive_face_detection_demo

Commit:
Fix "'err': unreferenced local variable" warning

aDanPin committed:
Update demos/interactive_face_detection_demo_gapi/main.cpp

Co-authored-by: Roman Donchenko <roman.donchenko@intel.com>

Commit:
Remove argument from catch block to avoid unnecessary logic and build warnings

Commit:
Got rid of unnecessary duplication of "setting source" code

Commit:
Fix for windows build

Commit:
Remove unnecessary comment

Commit:
Remove tralling whitespace

Commit:
Merge branch 'dp/gapi_interactive_face_detection_demo' of https://github.com/aDanPin/open_model_zoo into dp/gapi_interactive_face_detection_demo
